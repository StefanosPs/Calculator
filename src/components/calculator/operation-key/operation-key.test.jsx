import React from 'react';
import renderer from 'react-test-renderer';


import {
  mathOperationObj
} from "../../../classes/calculator/calculatorConstants";

import OperationKey from './operation-key';

it('renders mathOperationObj when there are no props', () => {
  const tree = renderer.create(<OperationKey />).toJSON();
  expect(tree).toMatchSnapshot(); 
});

it('renders mathOperationObj when there are props', () => {
    const numberKeyFull = renderer.create(<OperationKey mathOperationObj={mathOperationObj} />).toJSON();
    expect(numberKeyFull).toMatchSnapshot();
});