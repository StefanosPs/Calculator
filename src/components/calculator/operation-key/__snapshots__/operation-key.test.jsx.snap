// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders mathOperationObj when there are no props 1`] = `
<div
  className="container"
>
  <div
    className="justify-content-center row"
  >
    <div
      className="col"
    >
      <div
        className="fade alert alert-danger show"
        role="alert"
      >
        Prop mathOperationObj must be an object
      </div>
    </div>
  </div>
</div>
`;

exports[`renders mathOperationObj when there are props 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
        disabled={false}
        id="operation-key-divide"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        text="Divide"
        type="button"
      >
        /
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
        disabled={false}
        id="operation-key-multiplication"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        text="Μultiplication"
        type="button"
      >
        X
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
        disabled={false}
        id="operation-key-minus"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        text="Minus,"
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
        disabled={false}
        id="operation-key-plus"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        text="Add"
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
        disabled={false}
        id="operation-key-openParenthesis"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        text="Open Parenthesis"
        type="button"
      >
        (
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
        disabled={false}
        id="operation-key-closeParenthesis"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        text="Close Parenthesis"
        type="button"
      >
        )
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
        disabled={false}
        id="operation-key-squared"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        text="Squared"
        type="button"
      >
        x²
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
        disabled={false}
        id="operation-key-squareRoot"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        text="Square Root"
        type="button"
      >
        √x
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
        disabled={false}
        id="operation-key-undo"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        text="Undo"
        type="button"
      >
        ↶
      </button>
    </div>
    <div
      className="col-4"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-outline-{theme.color} btn-block"
        disabled={false}
        id="operation-key-clearDisplay"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        text="Clear Display"
        type="button"
      >
        C
      </button>
    </div>
    <div
      className="col-8"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <button
        className="shadow pl-3 pr-3 text-white mb-1 rounded btn btn-primary btn-block"
        disabled={false}
        id="operation-key-result"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        text="Calculate"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
